src/main/java/org/broadinstitute/hellbender/tools/walkers/varianteval/stratifications/AlleleFrequency.java

```diff
@@ -12,26 +13,74 @@
  */
+@DocumentedFeature(groupName= "tools", summary = "Stratify by  eval RODs by the allele frequency of the alternate allele") // , extraDocs = {VariantEval.class})
 public class AlleleFrequency extends VariantStratifier {
+
+    public enum StratifyingScale {
+        LINEAR,
+        LOGARITHMIC
+    }
+
+    private StratifyingScale scale;
+    private boolean useCompAFStratifier;
+
+    private static int logLimit = 30; // go from -30 to 30 using logit function
+
     @Override
     public void initialize() {
-        for( double a = 0.000; a <= 1.005; a += 0.005 ) {
-            states.add(String.format("%.3f", a));
+        scale = getVariantEvalWalker().getAFScale();
+        useCompAFStratifier = getVariantEvalWalker().getCompAFStratifier();
+
+        switch (scale) {
+            case LINEAR:
+                for (double a = 0.000; a <= 1.005; a += 0.005) {
+                    states.add(String.format("%.3f", a));
+                }
+                break;
+            case LOGARITHMIC:
+                for (int a = -logLimit; a <= logLimit; a += 1) {
+                    states.add(String.format("%d", a));
+                }
+
         }
     }
 
     public List<Object> getRelevantStates(ReferenceContext referenceContext, ReadsContext readsContext, FeatureContext featureContext, VariantContext comp, String compName, VariantContext eval, String evalName, String sampleName, String FamilyName) {
         if (eval != null) {
             try {
-                return Collections.singletonList((Object)String.format("%.3f", (5.0 * MathUtils.roundToNDecimalPlaces(eval.getAttributeAsDouble("AF", 0.0) / 5.0, 3))));
+                Double alleleFraction  = Collections.max(eval.getAttributeAsDoubleList("AF", 0.0));
+                if (useCompAFStratifier) {
+                    if (comp != null) {
+                        alleleFraction = Collections.max(comp.getAttributeAsDoubleList("AF", 0.0));
+                    } else {
+                        alleleFraction = 0.0; // any site that isn't in the comp should be the expected lowest allele fraction
+                    }
+                }
+                switch(scale) {
+                    case LINEAR:
+                        return Collections.singletonList((Object)String.format("%.3f", (5.0 * MathUtils.roundToNDecimalPlaces(alleleFraction / 5.0, 3))));
+                    case LOGARITHMIC:
+                        alleleFraction += Math.pow(10.0, -6); // never have AF of 0.0
+                        return Collections.singletonList(String.format("%d", getLogitBucket(alleleFraction)));
+                }
+
             } catch (Exception e) {
                 return Collections.emptyList();
             }
         }
 
         return Collections.emptyList();
     }
+
+    private int getLogitBucket(Double alleleFraction) {
+        // calculate logit value
+        Float score = (float)( -10 * Math.log10((alleleFraction/(1-alleleFraction))));
```

Comment:
Ok so I talked about this with Yossi, and I think we actually want to keep it like this; first because it means a bin width of 1 works really well, and he thought that the idea of phred-scale was actually beneficial. So I think I will keep it like this.

Reply:
OK. And this way seems to have binned the variants nicely so I guess there's no reason to change.